examples/pip_parse/WORKSPACE

```
@@ -41,6 +41,19 @@ pip_parse(
     # Here, we use the interpreter constant that resolves to the host interpreter from the default Python toolchain.
     python_interpreter_target = interpreter,
 
+    # Requirement groups allow Bazel to tolerate PyPi cycles by putting dependencies
+    # which are known to form cycles into groups together.
+    requirement_groups = {
```

I am not sure the `requirement_groups` terminology is the best. We could call them what they are - dependency-cycles but I am not sure what the best term could be.

---

I think `groups` is less opaque than the original `clusters`, but it'd be easy enough to tweak things to reference `cycles` explicitly.

---

What about:
```suggestion
    # The documentation for this could say that when 'pip' dependency cycles are
    # seen, they can use 'resolve_cycles' attribute to work-around this. In the
    # long term we should not need this argument.
    resolve_cycles = [
        [
            "sphinx",
            "sphincontrib-qthelp",
            "sphinxcontrib-htmlhelp"
            ...
        ],
    ]
```

The key argument is technically not needed because we could always get the first element of the `deps` list and then use that to construct a group name. For example, for this case we could have `sphinx` being used automatically.